lib/util/SortableSet.js

```diff
@@ -1,19 +1,32 @@
 "use strict";
-//TODO: Make this a generic type
-//https://github.com/Microsoft/TypeScript/issues/23385
-//https://github.com/Microsoft/TypeScript/issues/23384
+
+// eslint-disable-next-line valid-jsdoc
+/**
+ * A subset of Set that offers sorting functionality
+ * @template T item type in set
+ * @typedef {(a: T, b: T) => number} SortFunction
```

Comment:
If you have updated to a version of typescript that has Microsoft/TypeScript#23947 (from 17 May), `T` now belongs exclusively to `SortFunction`, not to `SortableSet`. `SortFunction` should be scoped to `SortableSet` anyway, so it should move to a comment in `SortableSet`'s body like on line 10 or so.